### DISCUSSION TITLE
supporting regular expressions

### DISCUSSION THREAD

**Discussion by sylvoslee** [gorules/zen#46] [General]
Will string expressions consider supporting regular expressions, or is there any other alternative

**Comment by ivanmiletic** [comment]
Hi @sylvoslee good suggestion, spoke with the team, and we will be adding regex support, at the moment priority will be median and mean (as seen here: https://github.com/gorules/zen/issues/45), and after that, we will add regex match

EDIT:
We have a few support functions that may cover some basic regex concepts: https://gorules.io/docs/rules-engine/expression-language/string. There you can find startsWith, endsWith, and contains.

Other than that you might put regex into a function block that chains to the decision table and returns true/false as a new field.

**Reply by david-AH** [reply]
Also very interested in this capability.  A large portion of our 'rules' rely on regular expressions.  Want to do something like the following, where the 'in' is applied as a regular expression matcher.

```
not some($, # in ["partialpattern*", "0210[a-z]"])
```

Alternatively, I could use startsWith IF I could do something like this:

```
not some($, startsWith(#, ["prefix1", "prefix2"]))
```

Is the latter possible?

**Reply by stefan-gorules** [reply]
Hi @david-AH I believe you can do something pretty close e.g.:

`not some($, startsWith(#, "prefix1") or startsWith(#, "prefix2"))`

just tried it with the expression node and seems to behave just fine. It might be better to use `none` instead of `not some` for readability.

`none(strings, startsWith(#, "prefix1") or startsWith(#, "prefix2"))`

**Comment by stefan-gorules** [comment]
Opened MR, let me know if there are some tests that you would like to see for the regex functions:

https://github.com/gorules/zen/pull/52

Unfortunately, the regex crate from Rust (as close to official as it gets) has some caveats (lack of look-around and backreferences).

**Reply by sjbodzo** [reply]
@stefan-gorules This makes sense. Look-around / backreferences can be very expensive, and in some cases open us up to a DOS style attack. Thank you for pushing this out!
